.github/workflows/pr-slack-notification.yaml

```diff
@@ -0,0 +1,287 @@
+            }")
+          fi
+
+          # Add closed PRs section if not empty
+          if [ -n "$closed_pr_list" ]; then
+            payload_blocks+=("{
+              \"type\": \"header\",
+              \"text\": {
+                \"type\": \"plain_text\",
+                \"text\": \"List of Pull Requests Closed Since the Last Cron Job\"
+              }
+            }")
+            payload_blocks+=("{
+              \"type\": \"section\",
+              \"text\": {
+                \"type\": \"mrkdwn\",
+                \"text\": $(echo "$closed_pr_list" | jq -sR .)
+              }
+            }")
+          fi
+
+          # Construct the payload
+          payload=$(jq -n --argjson blocks "$(printf '%s\n' "${payload_blocks[@]}" | jq -s '.')" '
+          {
+            "blocks": $blocks
+          }')
+
+          # Send the payload to Slack
+          curl -X POST -H 'Content-type: application/json' --data "$payload" $SLACK_WEBHOOK_URL || echo "Slack notification failed with status code: $?"
+
+      - name: Save current timestamp
+        run: |
+          current_timestamp=$(date -u +%Y-%m-%dT%H:%M:%SZ)
+          echo "$current_timestamp" > cache/last_run_timestamp.txt
+          echo "Current run timestamp saved: $current_timestamp"
+
+      - name: Verify Cache Save
+        if: always()
+        run: |
+          echo "Checking saved cache content...."
+          ls -l cache/
+          cat cache/notified_prs.json
+
+      - name: Save cache
+        if: always()
+        uses: actions/cache@v2
+        with:
+          path: cache
+          key: ${{ runner.os }}-pr-cache-${{ github.run_number }}
+
+      - name: Cleanup the restored cache key
+        run: |
+
+          # Fetching list of cache keys........
+          # Allowed values are within the limit 1-100
+          cacheKeys=$(gh actions-cache list -R $REPO -B $BRANCH -L 100 | cut -f 1 )
+          echo "Cache keys-->: $cacheKeys"
+
+          # Extract the most recent cache key
+          mostRecentCacheKey=$(echo "$cacheKeys" | head -n 1)
+          
+          # Setting this to not fail the workflow while deleting cache key
+          set +e
+          if [ -n "$mostRecentCacheKey" ] && [[ $mostRecentCacheKey == Linux-pr-cache-* ]]; then
+            echo "Deleting the most recent cache key..."
+            gh actions-cache delete $mostRecentCacheKey -R $REPO -B $BRANCH --confirm
+          else
+            echo "No cache keys found."
+          fi
+
+          echo "Done"
+
+    # 'secrets.GITHUB_TOKEN' is automatically provided by GitHub Actions for each workflow run. We don't need to manually create or manage this token.
+    # The below way, the workflow is granting write access to the actions scope of the GITHUB_TOKEN.
+    # If the below specific way is not successful to delete the cache key, then the repository owner will need to create a Personal Access Token (PAT) with write permissions, add it as a secret, and specify the name in GH_TOKEN.
```

Comment:
I am confused by the location of these comments

Reply:
@TrevCraw , I added the comment for the below code.
```
permissions:
      actions: write
```

I mentioned this in the box note that this code is responsible for deleting the `cache key`. If we do not specify it this way, the deletion of the cache key cannot happen, and we will need a PAT, which needs to be added as a secret for deletion.

<img width="756" alt="Screenshot 2024-06-29 at 11 03 15 AM" src="https://github.com/OpenLiberty/liberty-tools-intellij/assets/133869023/c63aea6c-7133-4f91-8837-a8d6164f1352">

For reference : https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#defining-access-for-the-github_token-scopes-1

If require, we can modify the comment.

Reply:
@vaisakhkannan Thank you for the explanation. We should make it clear that the write permission enables the cache key to be deleted. I would recommend the below modifications:
```suggestion
    # Below, the workflow is granting "write" access to the actions scope of the GITHUB_TOKEN. This enables the workflow to be able to delete cache keys.
    # If the "write" permission does not enable the ability to delete the cache keys, then the repository owner will need to create a Personal Access Token (PAT) with write permissions, add it as a secret, and then specify the name in GH_TOKEN.
```